### PR TITLE
19.add journal column

### PR DESCRIPTION
**Pull request by Redingston**

### CODE REVIEW

File: src/app/components/journal-column-item/journal-column-item.component.ts [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+        type: undefined
+      }
+      mark.pupilId = this.pupilShortInfo.id;
+      mark.value = value;
+      mark.type = this.testTypeId;
+      this.pupilLessonMarks.push(mark);
+    });
+    this.lessonMarks.emit(this.pupilLessonMarks);
+  }
+}
```
Comment (DmMov) [inline src/app/components/journal-column-item/journal-column-item.component.ts]: You need to explain all these lines to me.

File: src/app/components/lesson-item/lesson-item.component.html [hunk 2]
```diff
@@ -14,4 +14,6 @@ <h4 class="discipline-value">
 <div class="class-name">
    <span class="class-name-value">{{ item.className }}</span>
 </div>
-<button (click)="onClick()" class="btn start-btn">start lesson</button>
\ No newline at end of file
+<button (click)="onClick()" class="btn start-btn" [style.visibility]="visibleStart">start lesson</button>
+<button class="btn start-btn" [style.visibility]="visibleEdit">edit lesson</button>
+
```
Comment (DmMov) [inline src/app/components/lesson-item/lesson-item.component.html]: Dorou

File: src/app/components/lesson-item/lesson-item.component.ts [hunk 3]
```diff
@@ -5,14 +5,24 @@ import { Router } from '@angular/router';
 @Component({
   selector: 'yps-lesson-item',
   templateUrl: './lesson-item.component.html',
-  styleUrls: ['./lesson-item.component.scss']
+  styleUrls: ['./lesson-item.component.scss'],
 })
```
Comment (DmMov) [inline src/app/components/lesson-item/lesson-item.component.ts]: Nice change

File: src/app/components/lesson-item/lesson-item.component.ts [hunk 4]
```diff
@@ -5,14 +5,24 @@ import { Router } from '@angular/router';
...
   ngOnInit(): void {
+    if(this.item.isFinished==true){
+      this.visibleStart = "hidden";
+      this.visibleEdit = "show";
+    }
+    else{
+      this.visibleStart = "show";
+      this.visibleEdit = "hidden";
+    }
   }
```
Comment (DmMov) [inline src/app/components/lesson-item/lesson-item.component.ts]: Very strange logic

File: src/app/models/IPupil.ts [hunk 5]
```diff
@@ -10,3 +10,4 @@ export interface IPupil {
     imageUrl: string;
     schoolName: string;
 }
+
```
Comment (DmMov) [inline src/app/models/IPupil.ts]: Nice change too

File: src/app/pages/journal-column/journal-column.component.scss [hunk 6]
```diff
@@ -0,0 +1,72 @@
...
+    font-weight: bold;
+    background-color: $light;
+    border: 1px solid $accent;
+    padding: 0.5rem; 
+  }
+}
+.finish {
+  position: relative;
+  left: 89%;
+}
```
Comment (DmMov) [inline src/app/pages/journal-column/journal-column.component.scss]: The whole file is something weird, unstructured, copied and pasted. Also I think there are some useless styles.

File: src/app/pages/journal-column/journal-column.component.ts [hunk 7]
```diff
@@ -1,26 +1,90 @@
...
-    console.log(this.id + " - ID les " + this.classId + " - ID class " + this.teacherId + " - ID teacher");
+      this.http.post(apiUrl + "/JournalColumn", { topic: topic, lessonId: this.lessonId, classId: this.classId, lessonMarks: this.lessonMarks }, this.httpOtionsService.options)
+      .subscribe((successRes: any) => {
+        console.log('add journal column with id ', successRes);
+        this.router.navigate([`/teacher/lessons`]);
+      });
+    }
   }
-}
+}
```
Comment (DmMov) [inline src/app/pages/journal-column/journal-column.component.ts]: I can't even read that

**Review by DmMov — APPROVED** [review]

**Review by Hrabarchuk — APPROVED** [review]

**Review by Karonuk — APPROVED** [review]

**Pull request merged**